tests/unit/utils/test_validations.py

```diff
@@ -0,0 +1,683 @@
+# this script will test the validationmodel features
+from dataclasses import dataclass, field
+from datetime import date, datetime
+from enum import Enum
+from pathlib import Path
+from typing import Dict, List, Optional, Union
+
+import pytest
+from rdflib.term import Identifier
+
+from csvcubed.models.validatedmodel import (
+    ValidatedModel,
+    ValidationFunction,
+    Validations,
+)
+from csvcubed.models.validationerror import ValidateModelPropertiesError
+from csvcubed.utils import validations as v
+from csvcubed.utils.validations import (
+    validate_file,
+    validate_float_type,
+    validate_int_type,
+    validate_list,
+    validate_optional,
+    validate_str_type,
+    validate_uri,
+)
+from tests.unit.test_baseunit import get_test_cases_dir
+
+_test_case_base_dir = get_test_cases_dir() / "cli" / "inspect"
+
+
+class TestEnum(Enum):
+    First_Value = (0,)
+
+
+class OtherTestEnum(Enum):
+    First_Value = (0,)
+
+
+@dataclass
+class OtherTestClass(ValidatedModel):
+    str_test_variable_2: str
+
+    def _get_validations(self) -> Dict[str, ValidationFunction]:
+        return {"str_test_variable_2": validate_str_type}
+
+
+@dataclass
+class TestClass(ValidatedModel):
+    """This the test class that will be used in the tests below"""
+
+    str_test_variable: str = "Hello, World"
+    int_test_variable: int = 1
+    float_test_variable: float = 1.0
+    list_test_variable: List[str] = field(default_factory=list)
+    date_test_variable: Optional[date] = None
+    date_time_test_variable: Optional[datetime] = None
+    path_test_variable: Optional[Path] = None
+    test_uri: Optional[str] = None
+    test_enum_value: Optional[TestEnum] = None
+    test_any_of_value: Union[str, int, None] = None
+    test_validated_model_class: Optional[OtherTestClass] = None
+    test_data_type: Optional[str] = None
+    test_validate_instance_of: Optional[Identifier] = None
+
+    def _get_validations(self) -> Dict[str, ValidationFunction]:
+        return {
+            "str_test_variable": validate_str_type,
+            "int_test_variable": validate_int_type,
+            "float_test_variable": validate_float_type,
+            "list_test_variable": validate_list(validate_str_type),
+            "path_test_variable": validate_optional(validate_file),
+            "date_test_variable": validate_optional(v.date),
+            "date_time_test_variable": validate_optional(v.datetime),
+            "test_uri": validate_optional(validate_uri),
+            "test_enum_value": validate_optional(v.enum(TestEnum)),
+            "test_any_of_value": validate_optional(
+                v.any_of(validate_str_type, validate_int_type)
+            ),
+            "test_validated_model_class": validate_optional(
+                v.validated_model(OtherTestClass)
+            ),
+            "test_data_type": validate_optional(v.any_of(v.data_type, validate_uri)),
```

Comment:
This is the correct logic we want for validating data types, but needs to be applied to the actual data_type fields.

Reply:
Applied it to where data_type appears in observed_value.py